ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h

```diff
@@ -803,6 +800,15 @@ void RichardsMechanicsLocalAssembler<ShapeFunctionDisplacement,
                 displacement_index, displacement_index)
             .noalias() += B.transpose() * C * B * w;
 
+        auto const phi = _ip_data[ip].porosity;
+        double const p_FR = -chi_S_L * p_cap_ip;
+        // p_SR
+        variables[static_cast<int>(MPL::Variable::solid_grain_pressure)] =
+            p_FR - (sigma_eff + sigma_sw).dot(identity2) / (3 * (1 - phi));
```

Comment:
```suggestion
            p_FR - std::max(0.0, (sigma_eff + sigma_sw).dot(identity2) / (3 * (1 - phi)));
```

Reply:
or 
```
std::max(0.0, p_FR - (sigma_eff + sigma_sw).dot(identity2) / (3 * (1 - phi)));
```

Reply:
why?

Reply:
I do know what it is exactly the density model. If the density model describes the density change only under pressure, or compression, and no change in extension, all three principle stresses must be negative. If it is not the case, your implementation is roughly good, which means we do not care compressed or extended in directions but with a dominated status.

Reply:
The density model in the ctest is exponential one describing the change in both directions, compression and extension.